deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,121 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing a single dataset with a signle scalar output performance check."""
+import typing as t
+import torch
+from ignite.metrics import Metric, Accuracy
+from deepchecks.core import CheckResult, ConditionResult, DatasetKind
+from deepchecks.core.condition import ConditionCategory
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.vision.vision_data import TaskType
+from deepchecks.vision.metrics_utils.object_detection_precision_recall import \
+    ObjectDetectionAveragePrecision
+import warnings
+
+__all__ = ['SingleDatasetScalarPerformance']
+
+
+class SingleDatasetScalarPerformance(SingleDatasetCheck):
+    """Summarize a given metric over a given dataset to return a performance score as a scalar.
+
+    Parameters
+    ----------
+        metric : default: None
+        An ignite.Metric object whose score should be used. If None are given, use the default metrics.
+        reduce: torch function, default: None (for metrics that already return a scalar)
+        The function to reduce the scores vector into a single scalar
+    """
+    def __init__(self,
+                 metric: Metric = None,
+                 reduce: t.Callable = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.metric = metric
+        self.reduce = reduce
+
+    def initialize_run(self, context: Context, dataset_kind: DatasetKind.TRAIN):
+        """Initialize the metric for the check, and validate task type is relevant."""
+        if self.metric is None:
+            if context.train.task_type == TaskType.CLASSIFICATION:
+                self.metric = Accuracy()
+            elif context.train.task_type == TaskType.OBJECT_DETECTION:
+                self.metric = ObjectDetectionAveragePrecision()
+                self.reduce = torch.nanmean
+            else:
+                raise DeepchecksValueError('For task types other then classification or object detection, '
```

Comment:
Would consider a rephrase, maybe "No metric was passed. A metric must be passed for task types other than classification or object detection."